### PR TITLE
[BUGS-5879] Pass table values through our santization functions

### PR DESCRIPTION
**Pull request by jspellman814**

### CODE REVIEW

File: php/pantheon/utils.php [hunk 1]
```diff
@@ -60,8 +59,25 @@ public static function sanitize_data($data, $sanitizer_function = 'htmlspecialch
...
+			if ( 0 === $anchors->length ) {;
+				return $sanitizer_function($data);
+			}
+
+			$href = $anchors[0]->getAttribute('href');
+			$sanitized_href = call_user_func($sanitizer_function, $href);
+			$sanitized_link_text = call_user_func($sanitizer_function, $anchors[0]->textContent);
+			
+			// Rebuild anchor tags to ensure there are no injected attributes.
+			$rebuilt_link = '<a href="' . $sanitized_href . '">' . $sanitized_link_text . '</a>';
```
Comment (jspellman814) [inline php/pantheon/utils.php:79]: Need to confirm that this markup is accurate.

Comment (jspellman814) [reply php/pantheon/utils.php]: confirmed here: https://github.com/pantheon-systems/wp_launch_check/blob/b40dd826c2fac66f91a453355f4612ba847fcca0/php/pantheon/checks/plugins.php#L57

updated markup in [99f4652](https://github.com/pantheon-systems/wp_launch_check/pull/141/commits/99f4652be8b2dcfe91f8f211746e5330a9cf5661)

**Review by jazzsequence — COMMENTED** [review]
This looks like it will sanitize links but leaves other things alone. Since the initial bug report referenced injecting scripts into `img` tags, have you tested against that and would this address that possibility?

**Review by jazzsequence — CHANGES_REQUESTED** [review]
When I run this locally (I've pulled your changes into #142) I get a bunch of these PHP Warnings:

`PHP Warning:  DOMDocument::loadHTML(): htmlParseEntityRef: no name in Entity, line: 142 in /app/wp_launch_check/php/pantheon/utils.php on line 67`

**Comment by jspellman814** [comment]
> This looks like it will sanitize links but leaves other things alone. Since the initial bug report referenced injecting scripts into `img` tags, have you tested against that and would this address that possibility?

Strings that are not anchor tags are being passed through the sanitizer function, which was not previously happening: https://github.com/pantheon-systems/wp_launch_check/pull/141/files#diff-23eee0c3deeae33b0160199a3fa6283310be725332dde2dc19462ebbf87bfac2R72, though after testing it seems there is still work to do here. Also looking into the issue you ran into locally... I hadn't come across that when I was testing. =

File: php/pantheon/utils.php [hunk 2]
```diff
@@ -62,6 +62,26 @@ public static function sanitize_data($data, $sanitizer_function = 'htmlspecialch
...
 			return (string)$data;
+		} elseif ( is_string( $data ) ) {
+			if ( ! empty( $data ) ) {
+				$sanitized_data = $sanitizer_function($data);
+				$dom = new \DOMDocument;
+				$dom->loadHTML( $sanitized_data );
+				$anchors = $dom->getElementsByTagName('a');
+
+				// Bail if our string does not only contain an anchor tag.
+				if ( 0 === $anchors->length ) {;
```
Comment (pwtyler) [inline php/pantheon/utils.php:73]:
```suggestion
				if ( 0 === $anchors->length ) {
```
is this a typo? Valid just unconventional?

**Review by jazzsequence — APPROVED** [review]
Tested against the steps in the bugs ticket. WFM :shipit:

**Pull request merged**